tests/compute/global-generic-value-param.slang

```diff
@@ -0,0 +1,59 @@
+// global-generic-value-param.slang
+
+//TEST(compute):COMPARE_COMPUTE:
+
+// This is a basic test of support for global generic
+// value parameters: explicit named parameters at global
+// scope that can be used to generate specialized kernel
+// code based on different values.
+
+// We start by declaring a global generic value parameter:
+//
+// Note: only `int` parameters are expected to work for now.
+// Note: the default `= 0` intializer isn't used right now.
+//
+__generic_value_param kOffset : uint = 0;
+
+// For the test framework, we also need to specify what
+// value we want to specialize to.
+//
+// Note: this value (7) will be fed in to the compiler API
+// as a specialization argument, and will not be visible
+// to the compiler when it initially compiles the code
+// to IR.
+//
+//TEST_INPUT: globalSpecializationArg 7
```

Comment:
Being able to specify thing by name is obviously one of our long-term wish-list features for the render tests, and it would make sense for it to apply in this case too. I suspect that adding that feature would be best handled as an orthogonal change that just beefs up the test tool (leading to a lot of churn in existing test cases, I'm sure).

> Does this work because the __generic_value_param are in some well defined order (such as lexical order).

Yes, the order of `TEST_INPUT` lines for specialization parameters is significant, and the lexical order in the input file for each scope (global and entry-point) must match with the order of the parameters as reflected.

The alternative would be to have some kind of index operand to the `TEST_INPUT`s, but that would add clutter for now when we usually don't have more than one or two specialization parameters in a test anyway.